internal/server/server.go

```diff
@@ -81,6 +80,11 @@ type ListenerOptions struct {
 	Metrics string
 }
 
+type UIOptions struct {
+	Enabled  bool
```

Comment:
These should be exposed in the mapstructure for the config file. 

I think changing the structure here also makes this a breaking change.

Reply:
Good call out! It is a breaking change, I will call it out in the commit message!

I have a test showing this works from the config file in this PR, the test case name is `parse ui-proxy-url from config file`. Although it looks like I'm only testing the proxy url, not enabled as well. I'll update the test to show that both are working.

We shouldn't need the `mapstructure` struct field tags on every config field. Mapstructure actually compares fields names case-insensitive by default, so most of those tags are not doing anything. I think maybe the reason we needed some of them is because of `viper`? I'm planning on removing viper so that we don't have to add all these struct tags and things will work as expected. We can always add `mapstructure` struct tags if we want to customize something.